Description: Detect glibc's redirected posix_spawn_file_actions_addchdir declaration
Origin: upstream, https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=ca635799

diff -ur a/gettext-tools/gnulib-m4/posix_spawn_faction_addchdir.m4 b/gettext-tools/gnulib-m4/posix_spawn_faction_addchdir.m4
--- a/gettext-tools/gnulib-m4/posix_spawn_faction_addchdir.m4
+++ b/gettext-tools/gnulib-m4/posix_spawn_faction_addchdir.m4
@@ -1,5 +1,5 @@
 # posix_spawn_faction_addchdir.m4
-# serial 2
+# serial 3
 dnl Copyright (C) 2018-2026 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
@@ -11,10 +11,13 @@
   AC_REQUIRE([gl_SPAWN_H_DEFAULTS])
   AC_REQUIRE([AC_PROG_CC])
   gl_POSIX_SPAWN
-  AC_CHECK_FUNCS_ONCE([posix_spawn_file_actions_addchdir])
+  dnl Use AC_CHECK_DECLS, not AC_CHECK_FUNCS, because of glibc 2.44:
+  dnl <https://sourceware.org/bugzilla/show_bug.cgi?id=34437>.
+  AC_CHECK_DECLS([posix_spawn_file_actions_addchdir],,,
+    [[#include <spawn.h>]])
   gl_CHECK_FUNCS_ANDROID([posix_spawn_file_actions_addchdir_np],
     [[#include <spawn.h>]])
-  if test $ac_cv_func_posix_spawn_file_actions_addchdir = yes; then
+  if test $ac_cv_have_decl_posix_spawn_file_actions_addchdir = yes; then
     dnl This function is not yet standardized. Therefore override the
     dnl system's implementation always.
     REPLACE_POSIX_SPAWN_FILE_ACTIONS_ADDCHDIR=1
